lib/Site.js

```diff
@@ -185,6 +188,39 @@ Site.prototype.collectBaseUrl = function () {
   return Promise.resolve();
 };
 
+/**
+ * Collects the user defined variables map in the site/subsites
+ */
+Site.prototype.collectUserDefinedVariablesMap = function () {
+  // The key is the base directory of the site/subsites,
+  // while the value is user defined variable map for that particular subsite
+  this.userDefinedVariablesMap = {};
+
+  Object.keys(this.baseUrlMap).forEach((base) => {
+    const userDefinedVariables = {};
+    const definedVariablePath = path.resolve(base, USER_VARIABLES_PATH);
```

Comment:
Move this inside the `try` block since it's only used there, and rename it to `userDefinedVariablesPath`.